extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -482,8 +529,14 @@ export class AmpA4A extends AMP.BaseElement {
                   if (isValid) {
                     return creative;
                   }
-                  return Promise.reject(
-                      'Key failed to validate creative\'s signature.');
+                  // Only report is signature is expected to match given
+                  // multiple key providers could have been specified.
+                  if (!verifyHashVersion(signature, keyInfo)) {
+                    user().error(TAG, this.element.getAttribute('type'),
+                        `Key failed to validate creative\'s signature for
+                        service ${keyInfo.serviceName}.`, keyInfo.cryptoKey);
```

Comment:
That newline and leading whitespace won't be stripped, they'll appear in the error message.